include/nuttx/fs/fs.h

```diff
@@ -497,6 +546,8 @@ struct file_struct
   FAR struct file_struct *fs_next;      /* Pointer to next file stream */
   rmutex_t                fs_lock;      /* Recursive lock */
   int                     fs_fd;        /* File descriptor associated with stream */
+  FAR struct cookie_s     fs_cookie;    /* Cookie structure for fopencookie */
+  io_callbacks_t          fs_callbacks; /* Internal callbacks for IO call */
```

Comment:
duplicated with fs_cookie.cookie_io

Reply:
I would not put internal callbacks to the same structure as fopencookie callbacks simply for code clarity. Also we still need those structures twice... one for internal callback and other for user defined callback.

Reply:
Removed